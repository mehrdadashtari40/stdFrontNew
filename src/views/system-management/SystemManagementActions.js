import getAuthonticatedJSON from '../../common/utils/functions/getAuthenticatedJSON';

export const FETCH_CATEGORIES = 'FETCH_CATEGORIES';
export const FETCH_CATEGORY = 'FETCH_CATEGORY';
export const LOAD_USER = 'LOAD_USER';
export const LOAD_GROUPS = 'LOAD_GROUPS';
export const LOAD_DEPARTMENTS = 'LOAD_DEPARTMENTS';
export const LOAD_DEPARTMENT = 'LOAD_DEPARTMENT';



export function loadCategories(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setCategories(response));
		});
	};
}

export function setCategories(data) {
	return {
		type: FETCH_CATEGORIES,
		Categories: data,
	};
}

export function loadCategory(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setCategory(response));
		});
	};
}

export function setCategory(data) {
	return {
		type: FETCH_CATEGORY,
		Category: data,
	};
}

export function loadUser(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setUser(response));
		});
	};
}

export function setUser(data) {
	return {
		type: LOAD_USER,
		User: data,
	};
}

export function loadGroups(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setGroups(response));
		});
	};
}

export function setGroups(data) {
	return {
		type: LOAD_GROUPS,
		Groups: data,
	};
}

export function loadDepartments(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setDepartments(response));
		});
	};
}

export function setDepartments(data) {
	return {
		type: LOAD_DEPARTMENTS,
		Departments: data,
	};
}


export function loadDepartment(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setDepartment(response));
		});
	};
}

export function setDepartment(data) {
	return {
		type: LOAD_DEPARTMENT,
		Department: data,
	};
}


