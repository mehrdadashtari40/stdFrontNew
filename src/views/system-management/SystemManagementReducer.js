import {
	FETCH_CATEGORIES,
	FETCH_CATEGORY,
	LOAD_USER,
	LOAD_GROUPS,
	LOAD_DEPARTMENTS,
	LOAD_DEPARTMENT,
} from './SystemManagementActions';

const initialState = {
	Categories: [],
	Category: [],
	User: [],
	Users: [],
	Groups: [],
	Departments: [],
	Department: [],
};

export function systemManagementReducer(state = initialState, action) {
	switch (action.type) {
		case FETCH_CATEGORIES:
			return {
				...state,
				Categories: action.Categories,
			};

		case FETCH_CATEGORY:
			return {
				...state,
				Category: action.Category,
			};

		case LOAD_USER:
			return {
				...state,
				User: action.User,
			};
		case LOAD_GROUPS:
			return {
				...state,
				Groups: action.Groups,
			};
		case LOAD_DEPARTMENTS:
			return {
				...state,
				Departments: action.Departments,
			};
		case LOAD_DEPARTMENT:
			return {
				...state,
				Department: action.Department,
			};

		default:
			return state;
	}
}
