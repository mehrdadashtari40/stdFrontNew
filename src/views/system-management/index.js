import Categories from './components/setting/Categories';
import PmTables from './components/setting/PmTables';
import Backup from './components/backup/Backup';
import Setting from './components/setting/Setting';
import Users from './components/users/Users';
import Groups from './components/users/Groups';
import Department from './components/users/Departments';
export const routes = [
	{
		path: '/process-categories',
		exact: true,
		component: Categories,
		name: 'Categories',
	},
	{
		path: '/pm-tables',
		exact: true,
		component: PmTables,
		name: 'PmTables',
	},
	{
		path: '/backup',
		exact: true,
		component: Backup,
		name: 'Backup',
	},
	{
		path: '/setting',
		exact: true,
		component: Setting,
		name: 'Setting',
	},
	{
		path: '/users',
		exact: true,
		component: Users,
		name: 'Users',
	},
	{
		path: '/groups',
		exact: true,
		component: Groups,
		name: 'Groups',
  },
  
  {
		path: '/departments',
		exact: true,
		component: Department,
		name: 'Department',
	},
];
export * from './SystemManagementActions';
export * from './SystemManagementReducer';
