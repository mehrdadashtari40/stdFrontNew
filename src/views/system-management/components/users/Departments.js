import React from 'react';

import $ from 'jquery';
import getAuthonticatedJSON from '../../../../common/utils/functions/getAuthenticatedJSON';
import { config } from '../../../../config/config';
import putRequest from '../../../../common/utils/functions/putRequest';
import { smallBox, BigBreadcrumbs, WidgetGrid, JarvisWidget } from '../../../../common';
import postAuthenticatedJSON from '../../../../common/utils/functions/postAuthenticatedJSON';
import TreeView from '../../../../common/ui/components/TreeView';
import Form from '../../utils/DepartmentForm';
import DualListBox from 'react-dual-listbox';
import { connect } from 'react-redux';
import * as Actions from '../../SystemManagementActions';
import { withTranslation } from 'react-i18next';
import { compose } from 'redux';
import CardContent from "@mui/material/CardContent";
import Card from "@mui/material/Card";
import deleteAuthonticatedJSON from "../../../../common/utils/functions/deleteAuthonticatedJSON";
class Department extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			Departments: [],
			Department: [],
			Edit: false,
			Show: [],
			refresh: false,
			parent: '',
			AssignUsers: [],
			DepMembers: [],
			DepMembersIds: [],
			DepMembersIdsConst: [],
		};
		this.newHandelCreate = this.newHandelCreate.bind(this);
		this.handelIndex = this.handelIndex.bind(this);
		this.handelCreate = this.handelCreate.bind(this);
		this.handelShow = this.handelShow.bind(this);
		this.getAvailableUsers = this.getAvailableUsers.bind(this);
		this.assignUsers = this.assignUsers.bind(this);
		this.getAssignedUsers = this.getAssignedUsers.bind(this);
		this.assignUsers = this.assignUsers.bind(this);
		this.setDepartmentManager = this.setDepartmentManager.bind(this);
		this._filterTree = this._filterTree.bind(this);
	}

	componentDidMount() {
		let self = this;
		this.handelIndex();
		$('#DepartmentBox').on('hidden.bs.modal', function() {
			self.handelIndex();
			// document.getElementById('new-role-form').reset();
		});
		$(document).on('click', '.addDep', function(evt) {
			self.setState({ parent: evt.target.id });
			self.handelCreate(evt.target.id);
		});
		$(document).on('click', '.editDep', function(evt) {
			self.setState({ parent: evt.target.id });
			self.handelShow(evt.target.id);
		});
		$(document).on('click', '.deleteDep', function(evt) {
			self.setState({ parent: evt.target.id });
			self.handelShow(evt.target.id);
		});
		$(document).on('click', '.assignUser', function(evt) {
			self.setState({ parent: evt.target.id ? evt.target.id : '' });
			self.setState({ DepMembersIdsConst: self.state.DepMembersIds });
			self.getAssignedUsers();
			self.handleAssignUsers(evt.target.id);
		});
	}

	handelIndex() {
		getAuthonticatedJSON(config.apiServer + 'departments', {
			headers: {
				Authorization: 'Bearer ' + localStorage.getItem('access_token'),
				'Content-Type': 'application/json',
			},
		}).then(res => {
			res.map(r => {
				this.buildTree(r);
			});
			this.setState({ Departments: res });
			this.setState({ Department: res });
		});
		this.setState({ Edit: false });
	}

	handelCreate(parent_id) {
		this.enableFields();
		$('input[name=dep_parent]').val(parent_id);
		$('#actions').hide();
		$('#DepartmentBox').modal('show');
	}
	newHandelCreate() {
		this.enableFields();
		$('input[name=dep_parent]').val('');
		$('#actions').hide();
		$('#DepartmentBox').modal('show');
	}

	buildTree(data) {
		if (data !== null) data.children = data.dep_children;
		data.expanded = false;
		data.content = `<span class="node" id="${data.dep_uid}"><i class="fa fa-lg ${
			data.children.length === 0 ? `fa-angle-double-left` : `fa-folder-open`
		}"></i> + ${data.dep_title} (کاربران : ${data.dep_members})
                        <a class="addDep departman_nodebtn" ><i class="fa fa-plus" id="${data.dep_uid}"></i> </a>
                        <a class="editDep departman_nodebtn"><i class="fa fa-edit" id="${data.dep_uid}"></i></a>
                        <a class="assignUser departman_nodebtn"><i class="fa fa-users" id="${data.dep_uid}"></i></a>
                       ${
							data.dep_children.length > 0 || data.dep_members > 0
								? ''
								: `<a class="deleteDep departman_nodebtn"><i class="fa fa-remove" id="${data.dep_uid}"></i></a>`
						}
</span>`;
		data.children.map(r => {
			this.buildTree(r);
		});
	}

	handelShow(id) {
		this.props.loadDepartment(config.apiServer + 'department/' + id);
		getAuthonticatedJSON(config.apiServer + 'department/' + id, {
			headers: {
				Authorization: 'Bearer ' + localStorage.getItem('access_token'),
				'Content-Type': 'application/json',
			},
		}).then(res => {
			$('#actions').show();
			this.setState({ Show: res });
			$('#DepartmentBox').modal('show');
			this.SetData();
			this.disableFields();
		});
	}

	handleAssignUsers() {
		$('#AssignDepartmentUsers').modal('show');
	}

	assignUsers(selected) {
		let DepMembersIdsCon = this.state.DepMembersIds;
		let forAssign = selected.filter(data => {
			return DepMembersIdsCon.indexOf(data) < 0;
		});
		let forUnAssign = DepMembersIdsCon.filter(data => {
			return selected.indexOf(data) < 0;
		});
		this.setState({ DepMembersIds: selected });
		if (forAssign.length > 0) this.assignUser(forAssign);

		if (forUnAssign.length > 0) this.unAssignUser(forUnAssign);
	}

	setDepartmentManager() {
		let user = $('#manager').val();
		putRequest(config.apiServer + `department/${this.state.parent}/set-manager/${user}`, {
			headers: {
				Authorization: 'Bearer ' + localStorage.getItem('access_token'),
				'Content-Type': 'application/json',
			},
		}).then(res => {
			if (res.statusCode == 200) {
				smallBox({
					title: this.props.t('Successfully'),
					content: '<i>' + this.props.t('Data has been saved successfully') + '</i>',
					color: '#659265',
					iconSmall: 'fa fa-check fa-2x fadeInRight animated',
					timeout: 4000,
				});
				this.getAssignedUsers();
			}
		});
	}

	unSetDepartmentManager = () => {
		let user = $('#manager').val();
		var data = {
			USR_UID: user,
			DEP_UID: this.state.parent,
			NO_DEP_MANAGER: 0,
		};
		// postRequest(config.apiServer + `department/${this.state.parent}/set-manager/${user}`,
		postAuthenticatedJSON(config.apiServer + `departmentapi/unSetManager`, {
			body: JSON.stringify(data),
			headers: {
				Authorization: 'Bearer ' + localStorage.getItem('access_token'),
				'Content-Type': 'application/json',
			},
		}).then(res => {
			if (res.success) {
				smallBox({
					title: this.props.t('Successfully'),
					content: '<i>' + this.props.t('Data has been saved successfully') + '</i>',
					color: '#659265',
					iconSmall: 'fa fa-check fa-2x fadeInRight animated',
					timeout: 4000,
				});
				this.getAssignedUsers();
			}
		});
	};

	assignUser(users) {
		users.map(user => {
			postAuthenticatedJSON(config.apiServer + `department/${this.state.parent}/assign-user`, {
				body: JSON.stringify({ usr_uid: user }),
				headers: {
					Authorization: 'Bearer ' + localStorage.getItem('access_token'),
					'Content-Type': 'application/json',
				},
			}).then(res => {});
		});
		smallBox({
			title: this.props.t('Successfully'),
			content: '<i>' + this.props.t('Data has been saved successfully') + '</i>',
			color: '#659265',
			iconSmall: 'fa fa-check fa-2x fadeInRight animated',
			timeout: 4000,
		});
	}

	unAssignUser(unassigned) {
		try {
			unassigned.map(user => {
				deleteAuthonticatedJSON(config.apiServer + `department/${this.state.parent}/unassign-user/${user}`).then(res => {});
			});
			smallBox({
				title: this.props.t('Successfully'),
				content: '<i>' + this.props.t('Data has been saved successfully') + '</i>',
				color: '#659265',
				iconSmall: 'fa fa-check fa-2x fadeInRight animated',
				timeout: 4000,
			});
		} catch (e) {
		}
	}

	getAvailableUsers() {
		let self = this;
		getAuthonticatedJSON(config.apiServer + `department/${this.state.parent}/available-user`, {
			headers: {
				Authorization: 'Bearer ' + localStorage.getItem('access_token'),
				'Content-Type': 'application/json',
			},
		}).then(res => {
			let AssignUsers = [];
			let AssignUsersids = [];
			res.map(item => {
				AssignUsers.push({
					value: item.usr_uid,
					label: item.usr_firstname + ' ' + item.usr_lastname + ' ( ' + item.usr_username + ' )',
				});
				AssignUsersids.push(item.usr_uid);
			});
			if (self.state.DepMembers.length > 0)
				self.state.DepMembers.map(item => {
					AssignUsers.push(item);
				});

			self.setState({ AssignUsers: AssignUsers });
		});
	}

	getAssignedUsers() {
		let self = this;
		if (this.state.parent)
			getAuthonticatedJSON(config.apiServer + `department/${this.state.parent}/assigned-user`, {
				headers: {
					Authorization: 'Bearer ' + localStorage.getItem('access_token'),
					'Content-Type': 'application/json',
				},
			}).then(res => {
				let AssignedUsers = [];
				let AssignedUsersids = [];
				if (res.length == 0) {
					self.setState({ DepMembers: [] });
					self.getAvailableUsers();
				} else
					res.map(item => {
						AssignedUsers.push({
							value: item.usr_uid,
							label:
								item.usr_firstname +
								' ' +
								item.usr_lastname +
								(item.usr_supervisor ? ' * ' : ' ') +
								' ( ' +
								item.usr_username +
								' )',
							supervisor: item.usr_supervisor,
						});
						AssignedUsersids.push(item.usr_uid);
						self.setState({ DepMembers: AssignedUsers });
						self.setState({ DepMembersIds: AssignedUsersids });
						self.getAvailableUsers();
					});
			});
	}

	SetData() {
		let data = this.state.Show;
		$('input[name=dep_uid]').val(data['dep_uid']);
		$('input[name=dep_title]').val(data['dep_title']);
		$('input[name=dep_parent]').val(data['dep_parent']);
		$('select[name=dep_status]').val(data['dep_status']);
	}

	disableFields() {
		$('input[name=dep_title]').prop('disabled', true);
		$('input[name=dep_parent]').prop('disabled', true);
		$('select[name=dep_status]').prop('disabled', true);

		return false;
	}

	enableFields() {
		$('input[name=dep_title]').prop('disabled', false);
		$('input[name=dep_parent]').prop('disabled', false);
		$('select[name=dep_status]').prop('disabled', false);
		return false;
	}

	_filterTree(evt) {
		let cases = this.state.Department;
		if (evt.target.value != '') {
			let filtered = cases.filter(x => {
				return x.children.some(function(y) {
					if (y.children.length > 0) {
						return y.children.some(() => {
							return y.dep_title.includes(evt.target.value);
						});
					}
					return y.dep_title.includes(evt.target.value);
					// return y.name.includes(evt.target.value);
				});
			});
			this.setState({ Departments: filtered });
		} else this.setState({ Departments: cases });
	}

	render() {
		return (<>
			<Card id="content">
				<CardContent>
					<h5 className={'table-header-title'}>
						<i className={"fal fa-sitemap table-header-icon"}></i>
						{this.props.t('Departments List')}
						<button className="btn btn_inbox" onClick={()=>this.handelCreate()}>
							<i className="fa fa-plus"></i>
							<span>مورد جدید</span>
						</button>
					</h5>

					<div id="myTabContent1" className="tab-content">
						<TreeView items={this.state.Departments} role="tree" />
					</div>
				</CardContent>
			</Card>

			<div className="modal fade-in-up" id="DepartmentBox" role="dialog">
				<div className="modal-dialog modal-full">
					<div className="modal-content" id="DepartmentBoxBody">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-hidden="true">
								×
							</button>
							<h4 className="modal-title">تنظیمات دپارتمان</h4>
						</div>
						<Form
							Parent={this.state.parent}
							Refresh={this.handelIndex}
							Edit={this.state.Edit}
							Disable={this.disableFields}
							Enable={this.enableFields}
							Data={this.state.Show}
							Departments={this.state.Departments}
						/>
					</div>
				</div>
			</div>

			<div className="modal fade-in-up" id="AssignDepartmentUsers" role="dialog">
				<div className="modal-dialog modal-full">
					<div className="modal-content" id="AssignDepartmentUsersBody">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-hidden="true">
								×
							</button>
							<h4 className="modal-title">{this.props.t('Assign Users to Department')}</h4>
						</div>
						<div className="modal-body">
							{this.state.AssignUsers.length > 0 ? (
								<div className="row">
									<DualListBox
										canFilter
										filterCallback={(option, filterInput) => {
											if (filterInput === '') {
												return true;
											}

											return new RegExp(filterInput, 'i').test(option.label);
										}}
										filterPlaceholder={this.props.t('Search')}
										options={this.state.AssignUsers}
										selected={this.state.DepMembersIds}
										onChange={this.assignUsers}
									/>
									<div className="col-md-6">
										<select name="manager" id="manager" className="form-control">
											{this.state.DepMembers.map(x => {
												return (
													<option
														selected={x.supervisor ? `selected` : false}
														value={x.value}
													>
														{x.label}
													</option>
												);
											})}
										</select>
									</div>
									<div className="col-md-3">
										<button
											id="set"
											onClick={this.setDepartmentManager}
											className="btn btn-block btn-success"
										>
											{this.props.t('Set Manager')}
										</button>
									</div>
									<div className="col-md-3">
										<button
											id="unser"
											onClick={this.unSetDepartmentManager}
											className="btn btn-block btn-success"
										>
											{this.props.t('No Set Manager')}
										</button>
									</div>
								</div>
							) : (
								<div className="alert alert-danger">
									<p>کاربری برای انتصاب وجود ندارد</p>
								</div>
							)}
						</div>
					</div>
				</div>
			</div>
		</>)
	}
}
export default compose(
	withTranslation(),
	connect(state => state.systemManagement, Actions)
)(Department);
