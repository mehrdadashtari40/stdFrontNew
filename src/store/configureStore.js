import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';

import { userReducer, requestUserInfo } from '../common/user';
import { i18nReducer, i18nInit } from '../common/i18n';
import { trelloReducer } from '../views/trello';
import { systemManagementReducer } from '../views/system-management';
import { ProcessManagementReducer } from '../views/process-management';

import {reducer as layout2Reducer} from '../common/layout/_redux/layoutRedux'
import {reducer as tableReducer} from '../../src/common/tables/_redux/tableRedux'


import {reducer as basicInformationReducer} from '../views/basic-information/_redux/baseInfoRedux';
import LogsReducer from '../views/logs/LogReducesrs';

import BIReducer from '../views/BI/BIReducer';
import {reducer as inboxReducer} from "../views/inbox/_redux/inboxRedux";
import {reducer as authReducer} from '../../src/views/auth/_redux/authRedux'

import {reducer as ServiceDeskReducer} from "../views/service-desk/_redux/serviceDeskRedux";
import {reducer as SupervisorReducer} from "../views/supervisor/_redux/supervisorRedux";
import {reducer as CustomSettingsReducer} from "../views/customSettings/_redux/customSettingsRedux";
import {reducer as MenuReducer} from "../views/menu/_redux/menuRedux";
import {reducer as WorkDeskReducer} from "../views/work-desk/_redux/workDeskRedux";

export const rootReducer = combineReducers({
	auth: authReducer,
	user: userReducer,
	i18n: i18nReducer,
	basicInformation: basicInformationReducer,
	processManagement: ProcessManagementReducer,
	systemManagement: systemManagementReducer,
	ServiceDesk:ServiceDeskReducer,
	BI:BIReducer,
	trello:trelloReducer,
	inbox:inboxReducer,
	layout2:layout2Reducer,
	table:tableReducer,
	logs:LogsReducer,
	supervisor:SupervisorReducer,
	customSettingsReducer:CustomSettingsReducer,
	menuReducer:MenuReducer,
	workDeskReducer:WorkDeskReducer
});

const store = createStore(
	rootReducer,
	applyMiddleware(thunk)
);

store.dispatch(requestUserInfo());

store.dispatch(i18nInit());

export default store;
