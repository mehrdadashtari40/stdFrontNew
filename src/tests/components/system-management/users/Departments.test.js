import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../../store/configureStore';
import configureStore from 'redux-mock-store';
import * as selectActions from '../../../../views/system-management/SystemManagementActions';
import Departments from '../../../../views/system-management/components/users/Departments';
jest.dontMock('jquery');

const mockDepData = {
	dep_uid: '4887920315e248300ebfe31058662612',
	dep_parent: '',
	dep_title: 'A221rian',
	dep_status: 'ACTIVE',
	dep_manager: '',
	dep_manager: '6554024065d0d30a4395db8009405781',
	dep_manager_firstname: 'افشین',
	dep_manager_lastname: 'اسدنژاد',
	dep_manager_username: '2949498868',
	dep_members: 5,
	dep_parent: '',
	dep_status: 'ACTIVE',
	dep_title: 'A111rian',
	dep_uid: '5166284725e2481aba1ba99030294970',
	expanded: false,
	has_children: '1',
};

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));

test('Make Snapshot file , Departments Component', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(wrapper).toMatchSnapshot();
});

it('invokes `componentDidMount` when mounted', () => {
	jest.spyOn(Departments.prototype, 'componentDidMount');
	shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(Departments.prototype.componentDidMount).toHaveBeenCalled();
	Departments.prototype.componentDidMount.mockRestore();
});

// it('newHandelCreate HaveBeenCalled ?', () => {
// 	const wrapper = shallow(
// 		<Provider store={store}>
// 			<Departments />
// 		</Provider>
// 	)
// 		.dive({ context: { store } })
// 		.dive();
// 	const instance = wrapper.instance();
// 	const spy = jest.spyOn(instance, 'newHandelCreate');

// 	instance.forceUpdate();

// 	const Button = wrapper.find('.btn-outline-success');
// 	Button.simulate('click');
// 	expect(spy).toHaveBeenCalled();
// });

// it('setDepartmentManager HaveBeenCalled ?', () => {
// 	const wrapper = shallow(
// 		<Provider store={store}>
// 			<Departments />
// 		</Provider>
// 	)
// 		.dive({ context: { store } })
// 		.dive();
// 	const instance = wrapper.instance();
// 	const spy = jest.spyOn(instance, 'setDepartmentManager');

// 	instance.forceUpdate();

// 	const Button = wrapper.find('#set');
// 	Button.simulate('click');
// 	expect(spy).toHaveBeenCalled();
// });

// it('unSetDepartmentManager HaveBeenCalled ?', () => {
// 	const wrapper = shallow(
// 		<Provider store={store}>
// 			<Departments />
// 		</Provider>
// 	)
// 		.dive({ context: { store } })
// 		.dive();
// 	const instance = wrapper.instance();
// 	const spy = jest.spyOn(instance, 'unSetDepartmentManager');

// 	instance.forceUpdate();

// 	const Button = wrapper.find('#unset');
// 	Button.simulate('click');
// 	expect(spy).toHaveBeenCalled();
// });

test('handelIndex Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelIndex).toBeDefined();
});

test('handelCreate Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelCreate).toBeDefined();
});

test('newHandelCreate Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.newHandelCreate).toBeDefined();
});
test('buildTree Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.buildTree).toBeDefined();
});

test('handelShow Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelShow).toBeDefined();
});
test('handleAssignUsers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handleAssignUsers).toBeDefined();
});

test('assignUsers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.assignUsers).toBeDefined();
});

test('setDepartmentManager Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.assignUsers).toBeDefined();
});

test('unSetDepartmentManager Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.assignUsers).toBeDefined();
});

test('assignUser Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.assignUsers).toBeDefined();
});

test('getAvailableUsers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.getAvailableUsers).toBeDefined();
});

test('assignUsers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.assignUsers).toBeDefined();
});

test('getAssignedUsers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.getAssignedUsers).toBeDefined();
});

test('SetData Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.SetData).toBeDefined();
});

test('disableFields Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.disableFields).toBeDefined();
});

test('enableFields Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.enableFields).toBeDefined();
});

test('_filterTree Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Departments />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance._filterTree).toBeDefined();
});

describe('actions', () => {
	it('should create an action to loadDepartments', () => {
		const data = mockDepData;
		const expectedAction = {
			type: selectActions.LOAD_DEPARTMENTS,
			Departments: data,
		};
		expect(selectActions.setDepartments(data)).toEqual(expectedAction);
	});
});
