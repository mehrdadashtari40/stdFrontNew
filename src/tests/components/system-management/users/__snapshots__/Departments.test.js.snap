// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , Departments Component 1`] = `
<div
  id="content"
>
  <div
    className="row"
  >
    <BigBreadcrumbs
      className="col-xs-12 col-sm-7 col-md-7 col-lg-4"
      icon="fa fa-fw fa-Users"
      items={
        Array [
          "کاربران",
          "دپارتمان",
        ]
      }
    />
  </div>
  <WidgetGrid>
    <div
      className="row"
    >
      <article
        className="col-sm-12"
      >
        <JarvisWidget
          collapsed={false}
          color={false}
          colorbutton={false}
          custombutton={false}
          deletebutton={false}
          editbutton={false}
          fullscreenbutton={true}
          hidden={false}
          load={false}
          refresh={false}
          sortable={true}
          togglebutton={false}
        >
          <header>
            <span
              className="widget-icon"
            >
               
              <i
                className="fa fa-bars"
              />
               
            </span>
            <h2 />
            <div
              className="widget-toolbar"
            >
              <a
                className="btn btn-outline-success little_btn"
                onClick={[Function]}
              >
                <i
                  className="fa fa-plus"
                />
                <span />
              </a>
            </div>
          </header>
          <div
            className="tab-content"
            id="myTabContent1"
          >
            <div
              className="tab-pane fade in active"
            >
              <div
                className="row"
              >
                <div
                  className="col-sm-6 margin-top-10"
                >
                  <h2
                    className="h2"
                  >
                    <span
                      className="widget-icon"
                    >
                       
                      <i
                        className="fa fa-sitemap"
                      />
                       
                    </span>
                     
                     
                  </h2>
                </div>
              </div>
              <hr />
              <div
                className="tree smart-form"
              >
                <TreeView
                  items={Array []}
                  role="tree"
                />
              </div>
            </div>
          </div>
        </JarvisWidget>
      </article>
    </div>
  </WidgetGrid>
  <div
    className="modal fade-in-up"
    id="DepartmentBox"
    role="dialog"
  >
    <div
      className="modal-dialog modal-full"
    >
      <div
        className="modal-content"
        id="DepartmentBoxBody"
      >
        <div
          className="modal-header"
        >
          <button
            aria-hidden="true"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            ×
          </button>
          <h4
            className="modal-title"
          >
            تنظیمات دپارتمان
          </h4>
        </div>
        <DepartmentForm
          Data={Array []}
          Departments={Array []}
          Disable={[Function]}
          Edit={false}
          Enable={[Function]}
          Parent=""
          Refresh={[Function]}
          t={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="modal fade-in-up"
    id="AssignDepartmentUsers"
    role="dialog"
  >
    <div
      className="modal-dialog modal-full"
    >
      <div
        className="modal-content"
        id="AssignDepartmentUsersBody"
      >
        <div
          className="modal-header"
        >
          <button
            aria-hidden="true"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            ×
          </button>
          <h4
            className="modal-title"
          />
        </div>
        <div
          className="modal-body"
        >
          <div
            className="alert alert-danger"
          >
            <p>
              کاربری برای انتصاب وجود ندارد
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
